components/layouts/SetupLinkLayout.tsx

```diff
@@ -3,35 +3,50 @@ import Head from 'next/head';
 import Link from 'next/link';
 import Image from 'next/image';
 import { useRouter } from 'next/router';
-import Logo from '../../public/logo.png';
 import InvalidSetupLinkAlert from '@components/setup-link/InvalidSetupLinkAlert';
 import Loading from '@components/Loading';
 import useSetupLink from '@lib/ui/hooks/useSetupLink';
+import usePortalBranding from '@lib/ui/hooks/usePortalBranding';
+import { useTranslation } from 'next-i18next';
+import { hexToHsl, darkenHslColor } from '@lib/color';
+import { PoweredBy } from '@components/PoweredBy';
 
 export const SetupLinkLayout = ({ children }: { children: React.ReactNode }) => {
+  const { t } = useTranslation('common');
   const router = useRouter();
 
   const { token } = router.query as { token: string };
 
+  const { branding } = usePortalBranding();
   const { setupLink, error, isLoading } = useSetupLink(token);
 
   if (isLoading) {
     return <Loading />;
   }
 
+  const primaryColor = branding?.primaryColor ? hexToHsl(branding?.primaryColor) : null;
+  const title = setupLink?.service === 'sso' ? t('configure_sso') : t('configure_dsync');
```

Comment:
Getting a warning in the console
```Warning: Text content did not match. Server: "Configure Directory Sync" Client: "configure_dsync"```

Reply:
Updated `getStaticProps` to `getServerSideProps`, removed conditional spreading of locale.